### DISCUSSION TITLE
Anyway to detect sub processes in tasks?

### DISCUSSION THREAD

**Discussion by NORV-TECH** [WarmUpTill/SceneSwitcher#629] [Ideas]
Hi, it would be great if we can use a sub-process in tasks to set a rule. is that possible in future releases? thanks

**Comment by NORV-TECH** [comment]
![Screenshot 2022-12-11 115836](https://user-images.githubusercontent.com/67483516/206926090-b42276da-3e0a-4fe0-ab49-423eaf0370af.png)

**Comment by WarmUpTill** [comment]
I will try to look into it.

What exactly would be the use case?
Would it maybe be possible to simply combine a process condition with a "Window" condition?

**Reply by NORV-TECH** [reply]
Hi, thanks for the consideration.
In the example in the image above, I would like to select a "sub-process" to trigger a switch. This sub-process only appears when someone triggers the share screen and that's when I want to switch the scene.
Zoom also has many sub-processes, one for one a user share a doc, or switches views, shares audio.. etc.

I used Zoom as an example here but many programs have sub-processes specific to an action.

"Would it maybe be possible to simply combine a process condition with a "Window" condition?"
I am not sure.. I tried many things and couldnt find a way to switch the scene when I share is triggered.

**Reply by WarmUpTill** [reply]
After looking into it I think that this is already supported.
The issue here is just that the Task Manager is not showing the name of the _executable_ but instead its "Description" field.
So it displays "Zoom Sharing Host" instead of "CptHost.exe".

You can check the names of processes in the task manager by selecting the particular process, right clicking on it, and finally selecting "Properties".

![Zoom1](https://user-images.githubusercontent.com/19472752/208173648-b39087be-ab18-45a7-9e2f-2dd19dbc4c5d.png)

This will open up a dialog showing the name of the executable.

![Zoom2](https://user-images.githubusercontent.com/19472752/208173726-873a77bb-1f51-4686-8e0c-3268e01daa22.png)

Can you confirm that this also works for you?
